Kernel/Library/MiniStdLib.cpp

```diff
@@ -16,18 +16,16 @@ void* memcpy(void* dest_ptr, void const* src_ptr, size_t n)
     // FIXME: Support starting at an unaligned address.
     if (!(dest & 0x3) && !(src & 0x3) && n >= 12) {
         size_t size_ts = n / sizeof(size_t);
+        n -= size_ts * sizeof(size_t);
         asm volatile(
             "rep movsq\n"
```

Comment:
No `cld`?

Reply:
And why did you move the `n -= size_ts * sizeof(size_t);`?

Reply:
Not sure about the `cld`/iff we can make assumptions about `DF`

as for the code move, if I understand the `+` clobber correctly it means, in-out so it would clobber `size_ts`, make it 0, making the subtraction not work as intedet

Reply:
Oh yes, you are right about the inline assembly overriding `size_ts`.

The `cc` clobber should be unnecessary: https://stackoverflow.com/questions/75198561/how-to-let-gcc-know-that-the-direction-flag-df-in-the-eflags-register-has-change#answer-75203424

The x86-64 psABI only requires DF to be clear on function entry/exit, so I think it can technically be set when control flow reaches the inline asm.
Every example online seems to do a `cld` before instructions respecting the flag, but they never restore it to the previous state. But shouldn't that be wrong? Or does GCC expect DF to be zero after inline asm statements?

I could swear it didn't work without the `cld`, but it seems to work with your code.